File: Top_Interview_Questions/jumpGame.go
func canJump(nums []int) bool {
    if len(nums) == 1 {
        return true
    }
    cur := len(nums) - 1
    for i := cur - 1; i >= 0; i-- {
        if i + nums[i] >= cur {
            cur = i
        }
    }
    if cur == 0 {
        return true
    }
    return false
}

File: misc/jump_game.py
from typing import List

class Solution:
    def canJump(self, nums: List[int]) -> bool:
        goal = len(nums) - 1

        i = goal - 1

        while i >= 0:
            if i + nums[i] >= goal:
                goal = i

            i -= 1

        return goal == 0